Php81 compat

fritz-gerneth opened the pull request:
Had a few issues getting this to run:

1) composer.lock - could not install locked dependencies, not all used dependencies were present in composer.json. Rebuilt composer.json with all dependencies
2) PHPCS - Applied rules per phpcs
3) workflow - there are no php 8.(0/1)-zts-alpine images, cannot add to workflow

fritz-gerneth committed:
Update composer.json with all used dependencies, remove lock file

fritz-gerneth committed:
Apply phpcs rules

fritz-gerneth committed:
Add temporary / cache directories to .gitignore

WyriHaximus commented:
> 1. composer.lock - could not install locked dependencies, not all used dependencies were present in composer.json. Rebuilt composer.json with all dependencies

This is a known issue, there is one major blocker that blocks the remaining few dependencies from fully being 8 and 8.1 compatible and requireable. So that should be resolveable with a version bump of that tests utilities package in the future. Alternatively we can skip some checks for now.

> 2. PHPCS - Applied rules per phpcs

:+1: 

> 3. workflow - there are no php 8.(0/1)-zts-alpine images, cannot add to workflow

Switch to NTS images, this package doesn't need ZTS

fritz-gerneth committed:
Switch to nts & alpine images

fritz-gerneth commented:
Moved CI to NTS, remove the alpine version constraint as well.

fritz-gerneth committed:
Fix phpstan errors

fritz-gerneth committed:
Allow php >=7.4

fritz-gerneth committed:
Add php 8.0 & 8.1 to CI matrix

fritz-gerneth commented on .github/workflows/ci.yml:
```diff
@@ -6,11 +6,11 @@ jobs:
     strategy:
       matrix:
-        php: [7.4]
+        php: [7.4, 8.0, 8.1]
```
Not sure why, but for me the version for `8.0` is being cut off to `8`

WyriHaximus replied:
Put quotes around it, `8.0` is seen as a number that is shortened to `8`. Alternatively, you can look at doing this: https://github.com/WyriHaximus/php-test-utilities/commit/6c5fccfad363b84651402ef716409161d87b9ca9 with this flag enabled for `8.1`: https://github.com/WyriHaximus/php-test-utilities/blob/master/.github/workflows/ci.yml#L22-L23

fritz-gerneth replied:
Quotes did the trick

WyriHaximus commented:
> Moved CI to NTS, remove the alpine version constraint as well.

Hah awesome you picked up on that 🎉

WyriHaximus commented on .github/workflows/ci.yml:
```diff
@@ -30,13 +30,13 @@ jobs:
-        php: [7.4]
+        php: [7.4, "8.0", 8.1]
         composer: [lowest, current, highest]
         qa: [lint, cs, stan, psalm, unit-ci, infection, composer-require-checker, composer-unused]
```
```suggestion
        qa: [lint, cs, stan, psalm, unit, infection, composer-require-checker, composer-unused]
```

WyriHaximus commented on .github/workflows/ci.yml:
```diff
@@ -30,13 +30,13 @@ jobs:
   qa:
     strategy:
       matrix:
```
```suggestion
    strategy:
      fail-fast: false
      matrix:
```

WyriHaximus commented on .github/workflows/ci.yml:
```diff
@@ -6,11 +6,11 @@ jobs:
   composer-install:
     strategy:
       matrix:
```
```suggestion
    strategy:
      fail-fast: false
      matrix:
```

WyriHaximus commented:
Looking great so far, made two suggestions that will make sure we can see all jobs run and see all potential failures

fritz-gerneth commented:
I tried to update / fix dependencies where possible but I think I am at a dead end. Most dev dependencies are rather old and would need to be updated for PHP8.  For that an update of `wyrihaximus/test-utilities` is required. Unfortunately any version `>= 2.5` requires ` roave/backward-compatibility-check` which is not php8 compatible.

fritz-gerneth commented:
Figured the BC issue might be the reason you had the platform set to 7.4 in the composer file. Removed all dev dependencies from the composer file and updated the testing package. Now only infection is failing for 8.1 due to the lack of code coverage.

WyriHaximus reviewed (approved):
Awesome work, thank you so much for this :D! I Will update the test coverage over the weekend and release a new version.

WyriHaximus commented:
[`v1.1.0`](https://github.com/reactphp-inspector/metric/releases/tag/1.1.0) has just been released with these changes in it 🎉
